src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -481,7 +482,21 @@ private String getNameFromType(String type, boolean isArg) {
 			} else {
 				int count = nameCounts.compute(varName, (k, v) -> (v == null) ? 1 : v + 1);
 
-				return count == 1 ? varName : varName.concat(Integer.toString(count));
+				if (count > 1) {
+					varName = varName.concat(Integer.toString(count));
+				}
+				/*
+				 * Perform keyword check here, because appending numbers for duplicate names
+				 * should prevent keywords, so result will be:
+				 *   class_, class2, class3
+				 * If keyword check was performed before checking for duplicates, result would be:
+				 *   classs_, class_2, class_3
+				 */
+				else if (getInvalidJavaIdentifierReason(varName) == InvalidIdentifierReason.KEYWORD) {
+					varName += '_';
```

Comment:
This new name can clash with existing variables, like when user-provided through the mapping file. It needs to do the same nameCount operation as a few lines above and potentially fall back to not append _ and use the appropriate integer suffix instead.

On further look there appears to be a similar collision problem with pre-existing integer suffixed names (not generated by TR, but from mappings or original LVT) in the old code. The various `nameCounts.putIfAbsent` invocations that pre-populate the index with pre-existing names don't split those names into base-name and suffix. This way nameCounts may have an entry `(class2, 1)` instead of `(class, 2)` after recording `class2`. This then causes the name generator to turn `class` into `class_`, `class2`, `class3` instead of `class3`, `class4`, `class5`. I don't know if you want to fix this or if I should instead.

Reply:
Good point, should be fixed in the new version.

Not recording the name with the count suffix should be fixed now. However, I did not implement splitting of mapped names ending with a number. Not sure if it would really matter which variables got which number, and trying to determine the meaning of a number suffix could also behave incorrectly, e.g. when the class name ends with a number: `MyClass2`